pkg/controller/windowsmachine/windows/windows.go

```diff
@@ -216,6 +232,32 @@ func (vm *windows) ConfigureHybridOverlay(nodeName string) error {
 	return nil
 }
 
+func (vm *windows) ConfigureWindowsExporter() error {
+	// configures service to collect and expose metrics at endpoint with default port :9182 and default URL path /metrics
+	windowsExporterServiceArgs := "--collectors.enabled " +
+		"\"cpu,cs,logical_disk,net,os,service,system,textfile,container,memory\""
+
+	windowsExporterService, err := newService(WindowsExporterPath, windowsExporterServiceName, windowsExporterServiceArgs)
+	if err != nil {
+		return errors.Wrapf(err, "error creating %s service object", windowsExporterServiceName)
+	}
+	serviceExists, err := vm.serviceExists(windowsExporterServiceName)
+	if err != nil {
+		return errors.Wrapf(err, "error checking if %s Windows service exists", windowsExporterServiceName)
+	}
+	// create service if it does not exist.
+	if !serviceExists {
+		if err := vm.createService(windowsExporterService); err != nil {
+			return errors.Wrapf(err, "error creating %s Windows service", windowsExporterServiceName)
+		}
+	}
+	if err := vm.startService(windowsExporterService); err != nil {
+		return errors.Wrapf(err, "error starting %s Windows service", windowsExporterServiceName)
+	}
```

Comment:
I am seeing a pattern here. A helper function would be ideal that is used in `ConfigureKubeProxy()`. So introduce a helper, something like `ensureServiceIsRunning(serviceName)` and update `ConfigureKubeProxy()` to use that in a separate commit and then use it here.

Reply:
you can use that helper in the configureHybridOverlay method as well, the pattern is recurring in all three services.

Reply:
Good suggestion, thanks. I'll add another commit with this change.